pipeline/alignment.rules

```diff
@@ -92,15 +92,17 @@ rule convert_alignment_to_sorted_bam:
   output:
     join(WORKDIR, "{prefix}_aligned_coordinate_sorted.bam")
   params:
-    mem_gb = _mem_gb_for_ram_hungry_jobs()
+    mem_gb = _mem_gb_for_ram_hungry_jobs(),
+    tmpdir = join(WORKDIR, "{prefix}_tmp")
   benchmark:
     join(BENCHMARKDIR, "{prefix}_convert_alignment_to_sorted_bam.txt")
   log:
     join(LOGDIR, "{prefix}_convert_alignment_to_sorted_bam.log")
   resources:
     mem_mb = _mem_gb_for_ram_hungry_jobs() * 1024
   shell:
-    "picard -Xmx{params.mem_gb}g "
+    "TMPDIR={params.tmpdir} "
+    "picard -Xmx{params.mem_gb}g -Djava.io.tmpdir={params.tmpdir} "
```

Comment:
I'm mostly basing it on this: https://www.biostars.org/p/42613/

(likely just setting the TMPDIR is enough, but may as well set the Java internal one too just in case, not knowing enough about how picard works)